### ISSUE TITLE
Please add Swift package Manager support.

### ISSUE THREAD

**Issue by patchthecode** [braintree/braintree-ios-drop-in#155]
Please add Swift package Manager support.

**Comment by demerino** [comment]
This is on our roadmap and we will update when we are closer to adding support.

**Comment by JackoPlane** [comment]
@demerino Has there been any update on this?

**Comment by scannillo** [comment]
Hi @JackoPlane. 

We don't have much of an update at this point. Our SDK uses a few prebuilt binaries that we need to wait to be added to SPM before we can add support for it. We definitely plan to support Swift Package Manager in the future, though.

**Comment by thisolivier** [comment]
See similar issue in [braintree/braintree_ios #462](https://github.com/braintree/braintree_ios/issues/462 )

**Comment by callo90** [comment]
Any updates?

**Comment by sestevens** [comment]
@callo90 We're actively working on adding SPM support! We don't have a specific timeline quite yet, but we'll keep everyone posted through this issue.

**Comment by RamblinWreck77** [comment]
+1 on this, we're starting a new app and this is the only 3rd party lib we're planning to use that doesn't support SPM (thus requiring legacy cocoapods setup with the project).

We'd really love to keep the new project cocoapods free if at all possible :)

**Comment by RamblinWreck77** [comment]
@sestevens Any update on this? With Xcode 12/Swift 5.3's upgraded SPM now live I suspect SPM support will be much easier.

**Comment by scannillo** [comment]
Hi @RamblinWreck77 - we apologize for the delay, however wanted to update you that we are actively working on this across our iOS repos.

**Comment by RamblinWreck77** [comment]
Great news! Thanks for the quick reply!

Given that I really don't want to add cocoapods/carthage for a single dependency (especially with SPM support otw), is there a recommended way to add braintree manually to a SwitUI app in the mean time?

**Comment by scannillo** [comment]
@RamblinWreck77 - we don't currently offer official support for troubleshooting SwiftUI integrations. However you can refer to [this GH issue](https://github.com/braintree/braintree-ios-drop-in/issues/202).

**Comment by RamblinWreck77** [comment]
@scannillo sorry I should have been more clear, is there a recommended way to package braintree without cocoapods or carthage? (independent of UIKit or SwiftUI usage)

I'd really like to avoid cocoapods/carthage integration for a single dependency, it's so nice just having a simple xcode project + SPM (and compile time is faster too!)

**Comment by scannillo** [comment]
@RamblinWreck77 - no, we currently only recommend/support Carthage & CocoaPods integration types. You're welcome to manually integrate the SDK into your app, but we won't be able to help troubleshoot that.

**Comment by RamblinWreck77** [comment]
Just checking in, has there been any movement on SPM support the past 2 weeks?

**Comment by scannillo** [comment]
Hello - please refer to [this post](https://github.com/braintree/braintree_ios/issues/462#issuecomment-709594336) in the `braintree_ios` GitHub for an update on SPM support.

**Comment by lucaschoeneberg** [comment]
Is there any release date for the SPM function?

**Comment by sestevens** [comment]
@lucaschoeneberg We don't have a release date for SPM support in Drop-in quite yet. We just released SPM support in the [Braintree iOS SDK](https://github.com/braintree/braintree_ios/releases), and adding SPM support in Drop-in is high on our list of priorities.

**Comment by sestevens** [comment]
Hi everyone, we just released a beta version of the Drop-in SDK that includes Swift Package Manager support: https://github.com/braintree/braintree-ios-drop-in/releases/tag/9.0.0-beta1.

We recommend taking a look at our [Migration Guide](https://github.com/braintree/braintree-ios-drop-in/blob/master/V9_MIGRATION.md) if you want to check out this beta.

**Comment by sestevens** [comment]
We just released [v9 of the Drop-in SDK](https://github.com/braintree/braintree-ios-drop-in/releases/tag/9.0.0), which includes Swift Package Manager support.

Again, we recommend taking a look at our [Migration Guide](https://github.com/braintree/braintree-ios-drop-in/blob/master/V9_MIGRATION.md) for steps to update!

**Comment by JavierMazarioPicazo** [comment]
Is there any guide for Drop-in in Swift UI? I have already done the client side and a simple server but I don't know how to send the nonce to my server and test it

**Comment by scannillo** [comment]
@JavierMazarioPicazo - answered your question here (https://github.com/braintree/braintree-ios-drop-in/issues/266#issuecomment-832740962). This thread is regarding Swift Package Manager.
